pkg/ruler/compat.go

```diff
@@ -224,8 +227,29 @@ func RecordAndReportRuleQueryMetrics(qf rules.QueryFunc, queryTime prometheus.Co
 			logMessage := []interface{}{
 				"msg", "query stats",
 				"component", "ruler",
-				"cortex_ruler_query_seconds_total", querySeconds,
+			}
+			if origin := ctx.Value(promql.QueryOrigin{}); origin != nil {
+				queryLabels := origin.(map[string]interface{})
+				rgMap := queryLabels["ruleGroup"].(map[string]string)
+				logMessage = append(logMessage,
+					"rule_group", rgMap["name"],
+					"namespace", rgMap["file"],
+				)
+			}
+			ruleDetail := rules.FromOriginContext(ctx)
+			logMessage = append(logMessage,
+				"rule", ruleDetail.Name,
 				"query", qs,
+				"cortex_ruler_query_seconds_total", querySeconds,
+				"wall_time", queryStats.WallTime,
+				"fetched_series_count", queryStats.FetchedSeriesCount,
+				"fetched_chunk_count", queryStats.FetchedChunksCount,
+				"fetched_chunk_bytes", queryStats.FetchedChunkBytes,
+				"sample_count", queryStats.FetchedSamplesCount,
```

Comment:
Let's be consistent to https://github.com/cortexproject/cortex/blob/master/pkg/frontend/transport/handler.go#L325?

Make sure for available fields, we use the same field name.
Also, let's add fetched data bytes.

Reply:
Done